src/concurrent/files.rs

```diff
@@ -5,57 +5,58 @@ use crossbeam::channel::{Receiver, Sender};
 use serde::{Deserialize, Serialize};
 use tracing::debug;
 
+use crate::concurrent::Visit;
 use crate::grcov::covdir::Covdir;
 use crate::grcov::coveralls::Coveralls;
-use crate::utility::*;
+use crate::metrics::{get_covered_lines, get_root};
 use crate::{error::*, Complexity, Sort};
 
-use super::{ConsumerOutputWcc, Metrics, Wcc};
+use super::{get_cumulative_values, get_project_metrics, ConsumerOutputWcc, Metrics, Tree, Wcc};
 
-// Struct with all the metrics computed for a single file
+/// Struct with all the metrics computed for a single file
 #[derive(Clone, Default, Debug, Serialize, Deserialize, PartialEq)]
-pub(crate) struct FileMetrics {
-    pub(crate) metrics: Metrics,
-    pub(crate) file: String,
-    pub(crate) file_path: String,
+pub struct FileMetrics {
+    pub metrics: Metrics,
+    pub file: String,
+    pub path: String,
```

Comment:
Instead of using `String`, can we use `&Path` for path and for `file` a `Cow<str>` so we can avoid allocating? We can also remove the amount of `into` present in the code below